### PR TITLE
[OF-1198] feat: Add Stripe endpoints

### PR DESCRIPTION
**Pull request by Mag-JB**
* Add Customer Session Test
* Add Customer Portal Endpoint

connected-spaces-platform Pull Request

**For the reviewer**

* [ ] If required, are the changes covered by appropriate tests?
* [ ] Are any public-facing API changes well documented?
* [ ] Is the code easily readable and extensible and/or follow existing conventions?

### CODE REVIEW

**Review by MAG-SamBirley — APPROVED** [review]

**Review by MAG-SamBirley — APPROVED** [review]

File: Library/include/CSP/Systems/Users/UserSystem.h [hunk 1]
```diff
@@ -222,6 +225,15 @@ class CSP_API CSP_NO_DISPOSE UserSystem : public SystemBase
...
 												  NullResultCallback Callback);
 
+	/// @brief Get the Customer Portal Url for a user from Stripe
+	/// @param UserId csp::common::String : the id of the user associated with the customer portal
+	/// @param Callback StringResultCallback : callback that contains the customer portal URL of the User
+	CSP_ASYNC_RESULT void GetCustomerPortalUrl(const csp::common::String& UserId, CustomerPortalUrlResultCallback Callback);
+
+	/// @brief Get the checkout session Url for a user from Stripe
+	/// @param Tier csp::systems::TierNames : the tier of the checkout session needed
+	/// @param Callback CheckoutSessionUrlResultCallback : callback that contains the customer portal URL of the User
```
Comment (MAG-ThomasGreenhalgh) [inline Library/include/CSP/Systems/Users/UserSystem.h:235]: Looks like the description of `CheckoutSessionUrlResultCallback` is incorrect here.

File: Library/include/CSP/Systems/Users/UserSystem.h [hunk 2]
```diff
@@ -188,10 +189,12 @@ class CSP_API CSP_NO_DISPOSE UserSystem : public SystemBase
 	/// @brief Allow a user to reset their password if forgotten by providing an email address.
 	/// @param Email csp::common::String : account to recover password for
 	/// @param RedirectUrl csp::common::Optional<csp::common::String> : the URL to redirect the user to after they have registered
+	/// @param EmailLinkUrl csp::common::Optional<csp::common::String> : the URL inside the reset email sent to the user
 	/// @Param UseTokenChangePasswordUrl bool : if true the link in the email will direct the user to the Token Change URL
 	/// @param Callback NullResultCallback : callback to call when a response is received
 	CSP_ASYNC_RESULT void ForgotPassword(const csp::common::String& Email,
 										 const csp::common::Optional<csp::common::String>& RedirectUrl,
+										 const csp::common::Optional<csp::common::String>& EmailLinkUrl,
```
Comment (MAG-ThomasGreenhalgh) [inline Library/include/CSP/Systems/Users/UserSystem.h:197]: Is this considered a breaking change?

Comment (Mag-JB) [reply Library/include/CSP/Systems/Users/UserSystem.h:197]: yes 
image

**Review by MAG-AdamThorn — APPROVED** [review]

File: Tests/src/PublicAPITests/UserSystemTests.cpp [hunk 3]
```diff
@@ -1054,4 +1054,61 @@ CSP_PUBLIC_TEST(CSPEngine, UserSystemTests, AgeNotVerifiedTest)
 
 	LogOut(UserSystem);
 }
+#endif
+
+#if 0
+// Currently disabled whilst stripe testing is unavailable for OKO_TESTS
```
[the remark this answers is not in the document]
Comment (Mag-JB) [reply Tests/src/PublicAPITests/UserSystemTests.cpp:1060]: https://magnopus.atlassian.net/browse/OF-1204

**Review by MAG-AdamThorn — APPROVED** [review]

**Pull request merged**